### DISCUSSION TITLE
How do I convert the tag value?

### DISCUSSION THREAD

**Discussion by hrustiks** [frangoteam/FUXA#1503] [Q&A] [answered]
How do I convert the tag value? In the mqtt topic, the value is “ON” or “OFF”, but Fuxa only works with a number or a logical one. How to do it? I tried to use scripts - it doesn't work. And the documentation does not describe the difference between a read script and a write script. At what points does it work?
Please help me figure it out!

**Comment by MatthewReed303** [comment]
Convert the tag state to a string, you will need to create a string tag and use a script to read the number tag and write 'ON' or 'OFF' to the string tag

**Reply by hrustiks** [reply]
The task is reversed, but I understand you.
 However, I do not understand at what point (at what event) are the scripts called? 
 For example, the script selected as Write for the MQTT tag? Is it "Write" when the tag gets a value? Or something else? And what does the input parameter of the script “value" contain? The value received from mqtt?
 I tried to create a fuxa tag of the number type and change its value in the script (selected for both write and read) - the FUXA tag value does not change and there are no error messages in the logs. 
 Tell me what can be done and how?

**Reply by MatthewReed303** [reply]
The scale script is used to alter the value of that tag, not another tag value. You can create a server side script with 1 sec loop and use the get and set tag functions in there. Read the MQTT tag and if ON else if OFF else undefined and use set tag to write the value to Fuxa number tag

**Reply by hrustiks** [reply]
Thank you very much - I'll try again

**Comment by hrustiks** [comment]
It works with a script in a loop! Thanks for the hint!  
Is it possible to change the values coming to the tag from mqtt on the fly by scale script? I tried, but without success

**Comment by hrustiks** [comment, marked as answer]
The answer is there:   https://github.com/frangoteam/FUXA/issues/1533
